src/XMakeBuildEngine/Evaluation/Expander.cs

```diff
@@ -3080,6 +3080,23 @@ internal object Execute(object objectInstance, IPropertyProvider<T> properties,
                         args[0] = Convert.ChangeType(args[0], objectInstance.GetType(), CultureInfo.InvariantCulture);
                     }
 
+                    if (_receiverType == typeof(IntrinsicFunctions))
+                    {
+                        // Special case a few methods that take extra parameters that can't be passed in by the user
+                        //
+
+                        if (_methodMethodName.Equals("GetPathOfFileAbove"))
+                        {
+                            // Append the IElementLocation as a parameter to GetPathOfFileAbove
+                            //
+                            args = new []
+                            {
+                                args[0],
+                                elementLocation,
```

Comment:
Definitely don't love this but I also can't think of a better way at the moment.